core/src/Config.purs

```diff
@@ -479,25 +479,30 @@ legacyPackageSetEntryCodec = CA.object "LegacyPackageSetEntry"
   $ CA.recordProp (Proxy :: _ "dependencies") (CA.array PackageName.codec)
   $ CA.record
 
-readConfig :: forall a. FilePath -> Spago (LogEnv a) (Either String { doc :: YamlDoc Config, yaml :: Config })
+readConfig :: forall a. FilePath -> Spago (LogEnv a) (Either (Array String) { doc :: YamlDoc Config, yaml :: Config })
 readConfig path = do
   logDebug $ "Reading config from " <> path
   FS.exists path >>= case _ of
     false -> do
-      yml <- case map (_ <> ".yml") $ String.stripSuffix (String.Pattern ".yaml") path of
-        Nothing -> pure Nothing
-        Just y -> do
-          hasYml <- FS.exists y
-          pure $
-            if hasYml then
-              Just y
-            else
-              Nothing
-      let
-        message = case yml of
-          Nothing -> "Did not find " <> path
-          Just y -> "Did not find " <> path <> ". Spago's configuration files should end with .yaml. Try renaming " <> y
-      pure $ Left $ case path of
-        "spago.yaml" -> message <> " Run `spago init` to initialize a new project."
-        _ -> message
-    true -> liftAff $ FS.readYamlDocFile configCodec path
+      let replaceExt = map (_ <> ".yml") <<< String.stripSuffix (String.Pattern ".yaml")
+      yml <- map join $ for (replaceExt path) \yml -> do
+        hasYml <- FS.exists yml
+        pure $
+          if hasYml then
+            Just yml
+          else
+            Nothing
+      pure $ Left $ case path, yml of
+        "spago.yaml", Nothing -> 
+          [ "Did not find `" <> path <> "`. Run `spago init` to initialize a new project." ]
+        "spago.yaml", Just y ->
+          [ "Did not find `" <> path <> "`. Spago's configuration files must end with `.yaml`, not `.yml`. "
+          , "Try renaming `" <> y  <> "` to `" <> path <> "` or run `spago init` to initialize a new project."
+          ]
+        _, Nothing ->
+          ["Did not find `" <> path <> "`."]
+        _, Just y -> 
+            [ "Did not find `" <> path <> "`. Spago's configuration files must end with `.yaml`, not `.yml`. "
```

Comment:
This duplicates code, previously it was possible to have only one source line with this message

Reply:
Correct. But...

The message (per your original code) was outputting:
> Try renaming spago.yml Run `spago init` to initialize a new project.

I thought it would be better to put an `or` in between those two sentences when the context made sense. In some cases, one is likely the correct choice. And rather than write something weird like:
```
  fold $ Array.catMaybes
    [ Just ....
    , if cond1 then Just "Try renaming spago.yaml to..." else Nothing
    , if cond2 then Just "Run spago init" else Nothing
    , if cond3 then Just "Try ... or run ..." else Nothing
    ]
```
I thought it would be more readable to just duplicate the code. This isn't a performance-critical part of the codebase, so I don't think the duplication is harmful.

Reply:
Yeah I agree that this change is good